src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/AddDefendantLitigationFriendCallbackHandler.java

```diff
@@ -61,10 +74,120 @@ private CallbackResponse aboutToSubmit(CallbackParams callbackParams) {
+                        .respondent2LitigationFriendDate(currentDateTime)
+                        .respondent2LitigationFriendCreatedDate(
+                            ofNullable(callbackParams.getCaseData().getRespondent2LitigationFriendCreatedDate())
+                                .orElse(currentDateTime))
+                        .genericLitigationFriend(null);
+                } else if (caseData.getSelectLitigationFriend().getValue().getLabel().contains("Respondent One")) {
+                    caseDataUpdated
+                        .respondent1LitigationFriend(caseData.getGenericLitigationFriend())
+                        .respondent1LitigationFriendDate(currentDateTime)
+                        .respondent1LitigationFriendCreatedDate(
+                            ofNullable(callbackParams.getCaseData().getRespondent1LitigationFriendCreatedDate())
+                                .orElse(currentDateTime))
+                        .genericLitigationFriend(null);
+                }
+            } else {
+                caseDataUpdated
+                    .respondent1LitigationFriend(caseData.getGenericLitigationFriend())
+                    .respondent1LitigationFriendDate(currentDateTime)
+                    .respondent1LitigationFriendCreatedDate(
+                        ofNullable(callbackParams.getCaseData().getRespondent1LitigationFriendCreatedDate())
+                            .orElse(currentDateTime))
+                    .respondent2LitigationFriend(caseData.getGenericLitigationFriend())
+                    .respondent2LitigationFriendDate(currentDateTime)
+                    .respondent2LitigationFriendCreatedDate(
+                        ofNullable(callbackParams.getCaseData().getRespondent2LitigationFriendCreatedDate())
+                            .orElse(currentDateTime))
+                    .genericLitigationFriend(null);
+            }
+            return AboutToStartOrSubmitCallbackResponse.builder()
+                .data(caseDataUpdated.build().toMap(objectMapper))
+                .build();
+        }
+
+    }
+
     private SubmittedCallbackResponse buildConfirmation(CallbackParams callbackParams) {
         return SubmittedCallbackResponse.builder()
             .confirmationHeader("# You have added litigation friend details")
             .confirmationBody(exitSurveyContentService.respondentSurvey())
             .build();
     }
+
+    private CallbackResponse prepareDefendantSolicitorOptions(CallbackParams callbackParams) {
+        CaseData caseData = callbackParams.getCaseData();
+
+        List<String> dynamicListOptions = new ArrayList<>();
+        dynamicListOptions.add("Both");
+        dynamicListOptions.add("Respondent One: " + caseData.getApplicant1().getPartyName());
+
+        if (nonNull(caseData.getRespondent2())) {
+            dynamicListOptions.add("Respondent Two: " + caseData.getRespondent2().getPartyName());
+        }
+
+        isLitigationFriendForToBothParty(caseData);
```

Comment:
Not sure why this is required here?

Perhaps this method could be used by your if statement above to see if they have selected both - to determine to populate Respondent 1 + Respondent 2 fields